### PR TITLE
Change neighborhood from field to dropdown.

### LINKED ISSUE #708: On 'Add Listing' page, switch Neighborhood from Text Field To Dropdown
On the [Add Listing](https://sites-partners-tj3gg4i5eq-uc.a.run.app/listings/add) page, the Neighborhood field is currently a free-form text field. This creates the possibility of a neighborhood being added with an invalid name. To better validate the data, we should make this field a dropdown instead of a text field, with the dropdown populated by a valid neighborhood names.

add_listing_neighborhood

### PR DESCRIPTION
**Pull request by aisiri-murulidhar**
## Issue

- Closes #708
- Addresses #708
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Changes neighborhood selection from a textfield to a dropdown menu.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/addresses technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Provide instructions so we can review.

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.
Manual testing and cypress tests
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [x] I have run `yarn generate:client` if I made backend changes
- [ ] I have exported any new pieces in ui-components
Screen Shot 2021-11-23 at 11 32 38 AM
Screen Shot 2021-11-23 at 11 32 46 AM

### CODE REVIEW

File: shared-helpers/src/formKeys.ts [hunk 1]
```diff
@@ -178,3 +178,205 @@ export const preferredUnit = [
 ]
 
 export const bedroomKeys = ["studio", "oneBdrm", "twoBdrm", "threeBdrm"]
+
+export const neighborhoodKeys = [
```
Comment (plunkettgoogle) [inline shared-helpers/src/formKeys.ts:182]: Might be good to alphabetize this list.

File: ui-components/src/locales/general.json [hunk 2]
```diff
@@ -1235,6 +1235,206 @@
     "WI": "Wisconsin",
     "WY": "Wyoming"
   },
+  "neighborhoods":{
+    "Midtown": "Midtown",
```
Comment (plunkettgoogle) [inline ui-components/src/locales/general.json:1239]: Same comment here, could you alphabetize?

File: sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 3]
```diff
@@ -163,13 +163,22 @@ const BuildingDetails = ({
...
-        />
+        <ViewItem label={t("t.neighborhood")} className={"mb-0"}>
+          <Select
+            id={`neighborhood`}
+            name={`neighborhood`}
+            label={t("t.neighborhood")}
+            labelClassName="sr-only"
+            register={register}
+            controlClassName="control"
+            options={neighborhoodKeys}
```
Comment (plunkettgoogle) [inline sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx:174]: If you create a new listing, will the neighborhood be set to the JSON value, or the JSON key? The keys don't have spaces in them, but it looks like we're expecting the neighborhoods stored in the db to have spaces (i.e. human readable).

File: shared-helpers/src/formKeys.ts (hunk 1 above)
Comment (aisiri-murulidhar) [reply shared-helpers/src/formKeys.ts:182]: this list gets alphabetized with the sort function in the next line so in the dropdown the list is alphabetized, but should I still alphabetize it in the array as well for readability?

File: sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx (hunk 3 above)
Comment (aisiri-murulidhar) [reply sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx:174]: Looks like the JSON value will be saved

File: ui-components/src/locales/general.json (hunk 2 above)
Comment (aisiri-murulidhar) [reply ui-components/src/locales/general.json]: Done

File: ui-components/src/locales/general.json [hunk 4]
```diff
@@ -1235,6 +1235,205 @@
     "WI": "Wisconsin",
     "WY": "Wyoming"
   },
+  "neighborhoods": {"AirportSub": "Airport Sub",
```
Comment (plunkettgoogle) [inline ui-components/src/locales/general.json:1238]: Super nit: could you fix the indenting on this dictionary?

**Review by plunkettgoogle — APPROVED** [review]

File: ui-components/src/locales/general.json (hunk 4 above)
Comment (aisiri-murulidhar) [reply ui-components/src/locales/general.json:1238]: oops, fixed!

File: sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx [hunk 5]
```diff
@@ -163,13 +163,22 @@ const BuildingDetails = ({
...
-          label={t("t.neighborhood")}
-          name={"neighborhood"}
-          id={"neighborhood"}
-          placeholder={t("t.neighborhood")}
-          register={register}
-        />
+        <ViewItem label={t("t.neighborhood")} className={"mb-0"}>
+          <Select
+            id={`neighborhood`}
+            name={`neighborhood`}
```
Comment (GabeBenjamin) [inline sites/partners/src/listings/PaperListingForm/sections/BuildingDetails.tsx:169]: minor nit: I believe if you're only passing in strings for id and name can you just do:
```
id="neighborhood"
name="neighborhood"
```

File: ui-components/src/locales/general.json [hunk 6]
```diff
@@ -1235,6 +1235,206 @@
     "WI": "Wisconsin",
     "WY": "Wyoming"
   },
+  "neighborhoods": {
```
Comment (GabeBenjamin) [inline ui-components/src/locales/general.json:1238]: If we're not translating these, do we need this in the loacales json?

**Review by GabeBenjamin — APPROVED** [review]

File: ui-components/src/locales/general.json (hunk 6 above)
Comment (aisiri-murulidhar) [reply ui-components/src/locales/general.json]: Moved this to the building details form file.

**Pull request merged**